recipes/magnum-integration/all/conanfile.py

```diff
@@ -0,0 +1,165 @@
+        if self.options.shared:
+            del self.options.fPIC
+    
+    def requirements(self):
+        self.requires("magnum/{}".format(self.version))
+        if self.options.with_bullet:
+            self.requires("bullet3/3.17")
+        if self.options.with_dart:
+            # FIXME: Add 'dart' requirement
+            raise ConanInvalidConfiguration("DART library is not available in ConanCenter (yet)")
+        if self.options.with_eigen:
+            self.requires("eigen/3.4.0")
+        if self.options.with_glm:
+            self.requires("glm/0.9.9.8")
+        if self.options.with_imgui:
+            self.requires("imgui/1.84.2")
+        if self.options.with_ovr:
+            # FIXME: Add 'ovr' requirement
+            raise ConanInvalidConfiguration("OVR library is not available in ConanCenter (yet)")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_STATIC"] = not self.options.shared
+        self._cmake.definitions["BUILD_STATIC_PIC"] = self.options.get_safe("fPIC", False)
+        self._cmake.definitions["BUILD_TESTS"] = False
+        self._cmake.definitions["BUILD_GL_TESTS"] = False
+
+        self._cmake.definitions["WITH_BULLET"] = self.options.with_bullet
+        self._cmake.definitions["WITH_DART"] = self.options.with_dart
+        self._cmake.definitions["WITH_EIGEN"] = self.options.with_eigen
+        self._cmake.definitions["WITH_GLM"] = self.options.with_glm
+        self._cmake.definitions["WITH_IMGUI"] = self.options.with_imgui
+        self._cmake.definitions["WITH_OVR"] = self.options.with_ovr
+        
+        self._cmake.configure()
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def build(self):
+        self._patch_sources()
+
+        cm = self._configure_cmake()
+        cm.build()
+
+    def package(self):
+        cm = self._configure_cmake()
+        cm.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "MagnumIntegration"
+        self.cpp_info.names["cmake_find_package_multi"] = "MagnumIntegration"
+
+        lib_suffix = "-d" if self.settings.build_type == "Debug" else ""
+
+        if self.options.with_bullet:
+            self.cpp_info.components["bullet"].names["cmake_find_package"] = "Bullet"
+            self.cpp_info.components["bullet"].names["cmake_find_package_multi"] = "Bullet"
+            self.cpp_info.components["bullet"].libs = ["MagnumBulletIntegration{}".format(lib_suffix)]
+            self.cpp_info.components["bullet"].requires = ["magnum::magnum_main", "magnum::gl", "magnum::shaders", "bullet3::bullet3"]
+
+        if self.options.with_dart:
```

Comment:
No component or something is missing?

Reply:
I haven't been able to build with this option enabled. I prefer to leave it as TODO and add it once `dart` is available. I've added an Exception.

Reply:
Maybe `ConanException` instead of raw `Exception` ?